tests/test_resource_crawl.py

```diff
@@ -40,49 +40,108 @@ def test_empty_argument(self):
 class TestCrawlResourceIntegration(unittest.TestCase):
 
     def setUp(self):
-        self.server = ScrapyrtTestServer()
-        self.server.start()
-        self.crawl_url = self.server.url('crawl.json')
         self.site = MockServer()
         self.site.start()
         self.site_url = self.site.url('page1.html')
         self.spider_name = 'test'
+        self.server = ScrapyrtTestServer(site=self.site)
+        self.server.start()
+        self.crawl_url = self.server.url('crawl.json')
 
     def tearDown(self):
         if not self._passed:
             print self.server._non_block_read(self.server.proc.stderr)
         self.server.stop()
         self.site.stop()
 
+    def get_and_post(self, url, api_params, spider_data):
+        get_params = api_params.copy()
+        get_params.update(spider_data)
+        res_get = requests.get(
+            url, params=get_params
+        )
+        post_data = {
+            "request": spider_data
+        }
+        post_data.update(api_params)
+        res_post = requests.post(
+            url,
+            json=post_data
+        )
+        return res_get, res_post
+
     def test_no_parameters(self):
-        res = requests.get(self.crawl_url)
-        assert res.status_code == 400
-        res_json = res.json()
-        expected_result = {u'status': u'error',  u'code': 400}
-        self.assertDictContainsSubset(expected_result, res_json)
-        assert 'url' in res_json['message']
+        r1, r2 = self.get_and_post(self.crawl_url, {}, {})
+        for res in (r1, r2):
```

Comment:
I wonder whether it should be done by using [@pytest.mark.parametrize](https://pytest.org/latest/parametrize.html) - pass get and post functions as parameters, test body stays short and clean - both cases are tested